tests/PHPStan/Rules/Cast/data/invalid-cast.php

```diff
@@ -4,7 +4,7 @@ function (
 	string $str
 ) {
 	(string) $str;
-	(string) new \stdClass();
+	(string) new \stdClass(); // error: Cannot cast stdClass to string.
```

Comment:
what comes to mind: maybe we can do it similar to `assertType()` and `assertTypeCertainty()`

```suggestion
	assertErrorMessage('Cannot cast stdClass to string.', (string) new \stdClass());
```

?

Reply:
Yes, I'd like something like that too. Perhaps something like `assertErrorOnNextLine('message');`

Reply:
That is much less flexible. You cannot call method between class methods etc.

Reply:
Error can be anywhere, in phpdoc, at class level, at trait usage. Comment is the only thing you can place anywhere.

Reply:
Also, the implemented version works even within phpdocs (to be supported in https://github.com/phpstan/phpstan-src/pull/2807)